waku/waku_relay/protocol.nim

```diff
@@ -218,6 +244,24 @@ proc generateOrderedValidator*(w: WakuRelay): auto {.gcsafe.} =
     return ValidationResult.Accept
   return wrappedValidator
 
+proc validateMessage*(w: WakuRelay, pubsubTopic: string, msg: WakuMessage):
+  Future[Result[void, string]] {.async.} =
+    
+    for validator in w.wakuDefaultValidators:
+        let validatorRes = await validator(pubsubTopic, msg)
```

Comment:
Sorry but I can't quite understand why we need the "default validators" concept because we end up validating the message per its `pubsubTopic`

Reply:
My reasoning for this was first, to have exactly the same type of validator in the pubsub specific list and in the default validators list, using in both cases the type `WakuValidatorHandler`.

We could create a type `WakuDefaultValidatorHandler` without the pubsub topic parameter.
But I thought it's even better to have the pubsub topic as an argument. Notice that a default validator in this case is a validator that runs for all pubsub topics but doesn't need to behave the same way for all the topics. And that's the advantage of having the pubsub parameter in this case, it adds more possibilities to what we can (and might want to) do.

That's the reasoning behind it, not sure if it makes sense or if it just overcomplicates things

Reply:
> Notice that a default validator in this case is a validator that runs for all pubsub topics but doesn't need to behave the same way for all the topics

based on this i understand that the goal behind this "default validator" is to allow having different types of validators, the ones affecting to all pubusb topics, and the ones affecting to specific pubsub topics?

If that's the case, I would say there is no need to have this. A given validator shall be applied to all pubsub topics.

Reply:
mm I mean every validator is per topic. But my point is that we should enforce that whatever amount of validators are set, they are all applied to all subscribed topics.

I don't see a reason for having topica->validator_a, topicb->validator_b.

Reply:
> You mean that the same set of validators should run for all pubsub topics, meaning that each validator should be a "big proc" that handles the cases of all pubsub topics instead of validators being "small procs" specific to how to deal with a specific pubsub topic?

My point is that I don't see the need of having validators and default validators. Since IMHO every gossipsub topic that you are subscribed to, should apply the same set of validators.

> meaning that each validator should be a "big proc"

Note that from libp2p perspective, this is what we are doing. In nim-libp2p the order of execution of the validators is not deterministic (all of them are spawn async). So in waku we wrap them all into a single one, and we add that single one to nim-libp2p. In our case we do this because some validators consume more resources (eg rln one) so we don't want to execute that validator if previous one fail (eg decode into a WakuMessage).

Reply:
Got it! That makes lots of sense, didn't take into account that last point. So yes, no need of it. Thanks so much! 🤩